tests/foreman/cli/test_repository.py

```diff
@@ -1790,6 +1793,147 @@ def test_positive_upload_content_srpm(self):
+                         'Module Streams not synced correctly')
+
+        Repository.delete({'id': repo['id']})
+        with self.assertRaises(CLIReturnCodeError):
+            Repository.info({u'id': repo['id']})
+
+    @tier1
+    def test_module_stream_list_validation(self):
+        """Check module-stream get with list on hammer.
+
+         :id: 9842a0c3-8532-4b16-a00a-534fc3b0a776ff89f23e-cd00-4d20-84d3-add0ea24abf8
+
+         :Setup:
+             1. valid yum repo with Module Streams.
+         :Steps:
+             1. Create Yum Repositories with url contain module-streams and Products
+             2. Initialize synchronization
+             3. Verify the module-stream list with various inputs options
+
+         :expectedresults: Verify the module-stream list response.
+
+         :CaseAutomation: automated
+         """
+        repo1 = self._make_repository({
+            u'content-type': u'yum',
+            u'url': CUSTOM_MODULE_STREAM_REPO_1,
+        })
+        Repository.synchronize({'id': repo1['id']})
+        product2 = make_product_wait(
+            {u'organization-id': self.org['id']},
+        )
+        repo2 = self._make_repository({
+            u'content-type': u'yum',
+            u'url': CUSTOM_MODULE_STREAM_REPO_2,
+            u'product-id': product2['id']
+        })
+        Repository.synchronize({'id': repo2['id']})
+        module_streams = ModuleStream.list()
+        self.assertGreater(len(module_streams), 11,
+                           'Module Streams get worked correctly')
+        module_streams = ModuleStream.list({'product-id': product2['id']})
+        self.assertEqual(len(module_streams), 7,
+                         'Module Streams get worked correctly')
+
+    @tier1
+    def test_module_stream_info_validation(self):
+        """Check module-stream get with info on hammer.
+
+         :id: ddbeb49e-d292-4dc4-8fb9-e9b768acc441a2c2e797-02b7-4b12-9f95-cffc93254198
+
+         :Setup:
+             1. valid yum repo with Module Streams.
+         :Steps:
+             1. Create Yum Repositories with url contain module-streams
+             2. Initialize synchronization
+             3. Verify the module-stream info with various inputs options
+
+         :expectedresults: Verify the module-stream info response.
+
+         :CaseAutomation: automated
+         """
+        repo1 = self._make_repository({
+            u'content-type': u'yum',
+            u'url': CUSTOM_MODULE_STREAM_REPO_2,
+        })
+        Repository.synchronize({'id': repo1['id']})
+        product2 = make_product_wait(
+            {u'organization-id': self.org['id']},
+        )
+        repo2 = self._make_repository({
+            u'content-type': u'yum',
+            u'url': CUSTOM_MODULE_STREAM_REPO_2,
+            u'product-id': product2['id']
+        })
+        Repository.synchronize({'id': repo2['id']})
+        actual_result = ModuleStream.info({u'id': "5"})
```

Comment:
from where we got this id=5, perhaps we should list first with product-id to get the module steam id and verify here and later later with  that id

Reply:
@ldjebran I have updated the codebase with change.